libs/SmartSync/src/com/salesforce/androidsdk/smartsync/target/SoqlSyncDownTarget.java

```diff
@@ -98,21 +105,20 @@ private void addSpecialFieldsIfRequired() {
 	 */
 	public JSONObject asJSON() throws JSONException {
 		JSONObject target = super.asJSON();
-        target.put(QUERY, query);
+        if (query != null) target.put(QUERY, query);
 		return target;
 	}
 
     @Override
     public JSONArray startFetch(SyncManager syncManager, long maxTimeStamp) throws IOException, JSONException {
-        return startFetch(syncManager, maxTimeStamp, query);
+        return startFetch(syncManager, getQuery(maxTimeStamp));
     }
 
-    private JSONArray startFetch(SyncManager syncManager, long maxTimeStamp, String queryRun) throws IOException, JSONException {
-        String queryToRun = maxTimeStamp > 0 ? SoqlSyncDownTarget.addFilterForReSync(queryRun, getModificationDateFieldName(), maxTimeStamp) : queryRun;
-        RestRequest request = RestRequest.getRequestForQuery(syncManager.apiVersion, queryToRun);
+    protected JSONArray startFetch(SyncManager syncManager, String query) throws IOException, JSONException {
+        RestRequest request = RestRequest.getRequestForQuery(syncManager.apiVersion, query);
         RestResponse response = syncManager.sendSyncWithSmartSyncUserAgent(request);
-        JSONObject responseJson = response.asJSONObject();
-        JSONArray records = responseJson.getJSONArray(Constants.RECORDS);
+        JSONObject responseJson = getResponseJson(response);
+        JSONArray records = getRecordsFromResponseJson(responseJson);
```

Comment:
Extraction of records is moved to a separate method.
In the case of ParentChildrenSyncDownTarget, we are getting parent and children records together and we needed some control over the parsing without having to override the entire method,